hbase_storage/src/main/java/com/splicemachine/storage/HNIOFileSystem.java

```diff
@@ -197,6 +194,24 @@ public HFileInfo(FileStatus fileStatus) {
             return res;
         }
 
+        @Override
+        public FileInfo[] listDir()
+        {
+            if( !exists() ) return new FileInfo[] {};
+            if( !isDirectory() ) return new FileInfo[] { this };
+            ArrayList<FileInfo> res = new ArrayList<>();
+            try {
+                RemoteIterator<LocatedFileStatus> iterator = fs.listLocatedStatus(path);
+                while (iterator.hasNext()) {
+                    res.add( new HFileInfo(iterator.next()) );
+                }
+
+            } catch (IOException e) {
+                return new FileInfo[0];
```

Comment:
Shouldn't we at least log the exception (or re-throw it)? otherwise the user might think the directory was actually empty but in reality we e.g. couldn't connect.

Reply:
done